engine/src/main/java/com/datatorrent/stram/StreamingAppMasterService.java

```diff
@@ -880,6 +883,20 @@ private void execute() throws YarnException, IOException
         if (0 != exitStatus) {
           if (allocatedContainer != null) {
             numFailedContainers.incrementAndGet();
+            if (exitStatus != 1) {
+              // If container failure due to framework
+              AtomicInteger failedTimes = new AtomicInteger(1);
+              String hostname = allocatedContainer.container.getNodeId().getHost();
+              if (failedContainersMap.containsKey(hostname)) {
+                failedTimes = failedContainersMap.get(hostname);
+                int failureOnNode = failedTimes.incrementAndGet();
```

Comment:
If later we decide to track not only counter of failed attempts and add for example time of the last failed attempt or keep the full list of failed attempts, keeping the entry is preferable. It also leads to less garbage collection and because the map will not too big it should be OK to keep the entry and reset the counter to 0.